pyanaconda/packaging/__init__.py

```diff
@@ -428,18 +431,51 @@ def _getTreeInfo(self, url, proxy_url, sslverify):
                          proxy_url, e)
 
         ug = URLGrabber()
-        try:
-            treeinfo = ug.urlgrab("%s/.treeinfo" % url,
-                                  "/tmp/.treeinfo", copy_local=True,
-                                  proxies=proxies, **ugopts)
-        except URLGrabError as e:
+
+        # Retry treeinfo downloads with a progressively longer pause,
+        # so NetworkManager have a chance setup a network and we have
+        # full connectivity before trying to download things. (#1292613)
+        xdelay = xprogressive_delay()
+        for retry_count in xrange(0, MAX_TREEINFO_DOWNLOAD_RETRIES + 1):
```

Comment:
This code is borrowed from this commit https://github.com/rhinstaller/anaconda/commit/8c2544c6f537240179569d0068a7b22250e21a25. But from my understanding the first is not retry so that is why 0. +1 because the xrange is open interval from the end.